src/intro_gui.cpp

```diff
@@ -33,13 +39,207 @@
+	{
+		if (this->vehicle != INVALID_VEHICLE) {
+			const Vehicle *v = Vehicle::Get(this->vehicle);
+			this->position = RemapCoords(v->x_pos, v->y_pos, v->z_pos);
+		}
+
+		Point p;
+		switch (this->align_h) {
+			case LEFT: p.x = this->position.x; break;
+			case CENTRE: p.x = this->position.x - vp->virtual_width / 2; break;
+			case RIGHT: p.x = this->position.x - vp->virtual_width; break;
+		}
+		switch (this->align_v) {
+			case TOP: p.y = this->position.y; break;
+			case MIDDLE: p.y = this->position.y - vp->virtual_height / 2; break;
+			case BOTTOM: p.y = this->position.y - vp->virtual_height; break;
+		}
+		return p;
+	}
+};
+
+
 struct SelectGameWindow : public Window {
+	/** Vector of viewport commands parsed. */
+	std::vector<IntroGameViewportCommand> intro_viewport_commands;
+	/** Index of currently active viewport command. */
+	size_t cur_viewport_command_index;
+	/** Time spent (milliseconds) on current viewport command. */
+	uint cur_viewport_command_time;
+
+	/**
+	 * Find and parse all viewport command signs.
+	 * Fills the intro_viewport_commands vector and deletes parsed signs from the world.
+	 */
+	void ReadIntroGameViewportCommands()
+	{
+		intro_viewport_commands.clear();
+
+		/* Regular expression matching the commands: T, spaces, integer, spaces, flags, spaces, integer */
+		const char *sign_langauge = "^T\\s*([0-9]+)\\s*([-+A-Z0-9]+)\\s*([0-9]+)";
+		std::regex re(sign_langauge, std::regex_constants::icase);
+
+		/* List of signs successfully parsed to delete afterwards. */
+		std::vector<SignID> signs_to_delete;
+
+		for (const Sign *sign : Sign::Iterate()) {
+			std::smatch match;
+			if (std::regex_search(sign->name, match, re)) {
+				IntroGameViewportCommand vc;
+				/* Sequence index from the first matching group. */
+				vc.command_index = std::stoi(match[1].str());
+				/* Sign coordinates for positioning. */
+				vc.position = RemapCoords(sign->x, sign->y, sign->z);
+				/* Delay from the third matching group. */
+				vc.delay = std::stoi(match[3].str()) * 1000; // milliseconds
+
+				/* Parse flags from second matching group. */
+				enum IdType {
+					ID_NONE, ID_VEHICLE
+				} id_type = ID_NONE;
+				for (char c : match[2].str()) {
+					if (isdigit(c)) {
+						if (id_type == ID_VEHICLE) {
+							vc.vehicle = vc.vehicle * 10 + (c - '0');
+						}
+					} else {
+						id_type = ID_NONE;
+						switch (toupper(c)) {
+							case '-': vc.zoom_adjust = +1; break;
+							case '+': vc.zoom_adjust = -1; break;
+							case 'T': vc.align_v = IntroGameViewportCommand::TOP; break;
+							case 'M': vc.align_v = IntroGameViewportCommand::MIDDLE; break;
+							case 'B': vc.align_v = IntroGameViewportCommand::BOTTOM; break;
+							case 'L': vc.align_h = IntroGameViewportCommand::LEFT; break;
+							case 'C': vc.align_h = IntroGameViewportCommand::CENTRE; break;
+							case 'R': vc.align_h = IntroGameViewportCommand::RIGHT; break;
+							case 'P': vc.pan_to_next = true; break;
```

Comment:
Open question: should this pan to the next, or should it pan towards this sign?

The reason I ask, I keep wondering how I would use this system. And I would put a sign down, and go: okay, if you get here, how should that happen? Via a vehicle? Via panning?

In that sense, it feels like this is in the wrong direction. But I do not mind this at all, so it really is an open question.

Reply:
The other way of viewing it: What happens when this command activates? Does it zoom in/out? Does it start following a vehicle? Does it start panning? If panning required a flag on the _following_ command, then that would differ from the other flags in that you had to look at the next command to know when panning started.